Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -0,0 +1,94 @@
+                for (var i = 0; i < vmssList.length; i++) {
+                    if(vmssList[i].name.toUpperCase() === this.taskParameters.vmssName.toUpperCase())
+                    {
+                        resourceGroupName = utils.getResourceGroupNameFromUri(vmssList[i].id);
+                        osType = vmssList[i].properties.virtualMachineProfile.storageProfile.osDisk.osType;
+                        break;
+                    }
+                }
+
+                if(!resourceGroupName) {
+                    return reject(tl.loc("FailedToGetRGForVMSS", this.taskParameters.vmssName));
+                }
+
+                switch (this.taskParameters.action) {
+                    case "UpdateImage":
+                    var extensionMetadata: azureModel.VMExtensionMetadata = null;
+                    if(!!this.taskParameters.customScriptUrl) {
+                        extensionMetadata = this._getCustomScriptExtensionMetadata(osType);
+                        var customScriptExtension: azureModel.VMExtension = {
+                            name: "CustomScriptExtension" + Date.now().toString(),
+                            properties: {
+                                type: extensionMetadata.type,
+                                publisher: extensionMetadata.publisher,
+                                typeHandlerVersion: extensionMetadata.typeHandlerVersion,
+                                autoUpgradeMinorVersion: true,
+                                settings: {
+                                    "fileUris": [ this.taskParameters.customScriptUrl ],
+                                    "commandToExecute": this.taskParameters.customScriptCommand
+                                }
+                            }
+                        }
+                    }
+
+                    client.virtualMachineScaleSets.updateImage(resourceGroupName, this.taskParameters.vmssName, this.taskParameters.imageUrl, customScriptExtension, null, (error, result, request, response) => {
+                        if (error) {
+                            return reject(tl.loc("VMSSImageUpdateFailed", utils.getError(error)));
+                        }
+                        console.log(tl.loc("UpdatedVMSSImage"));
+                        resolve();
+                    });
+                    break;
+                }
+            });
+        });
+    }
+
+    private _getCustomScriptExtensionMetadata(osType): azureModel.VMExtensionMetadata {
+        if(osType === "Windows") {
+            return <azureModel.VMExtensionMetadata>{
+                type: "CustomScriptExtension",
+                publisher: "Microsoft.Compute",
+                typeHandlerVersion: "1.8"
+            }
+        } else if(osType === "Linux") {
+            return <azureModel.VMExtensionMetadata>{
+                type: "CustomScriptForLinux",
+                publisher: "Microsoft.OSTCExtensions",
+                typeHandlerVersion: "1.5"
```

Comment:
we can't say 1.*, but we can say 1.0 which means latest 1. version will be used. I will just check.